### PR TITLE
Remove env_logger and lower MSRV.

### PR DESCRIPTION
**Pull request by CosminPerRam**
Late edit: This PR was initially made to update env_logger to resolve an audit warning, doing so would have increased the MSRV to 1.63 but just for running the tests which included compiling the examples, which just using the library, isn't necessary, it was decided that we can just remove it, which also concluded that MSRV is actually 1.60, not the previously declared 1.61

[Cargo audit](https://crates.io/crates/cargo-audit) is a tool that audits your dependencies for crates with security vulnerabilities reported to the [RustSec Advisory Database](https://github.com/RustSec/advisory-db/).

Running `cargo audit` before resulted in:
```
Scanning Cargo.lock for vulnerabilities (58 crate dependencies)
Version:   0.2.14
Warning:   unsound
Title:     Potential unaligned read
Date:      2021-07-04
URL:       https://rustsec.org/advisories/RUSTSEC-2021-0145
Dependency tree:
atty 0.2.14
└── env_logger 0.9.3
    └── mdns-sd 0.8.0

warning: 1 allowed warning found
```

Updating env_logger to `0.10` fixes this and it doesn't introduce any breaking changes.

### CODE REVIEW

**Comment by CosminPerRam** [comment]
Ah, apparently `1.63` is required, unless raising MSRV, this cannot be done.

**Comment by keepsimple1** [comment]
> Ah, apparently `1.63` is required, unless raising MSRV, this cannot be done.

Sure, please raise MSRV to 1.63.  (It's surprising that `env_logger` would bring in `rustix`, luckily it's only used in tests).

**Comment by CosminPerRam** [comment]
MSRV of the crate (excluding dev depends) is actually 1.60, not 1.61 (and with dev its 1.63). This has been found using [cargo-msrv](https://github.com/foresterre/cargo-msrv) (running `cargo msrv` returns 1.60 and `cargo msrv -- cargo t` returns 1.63 (minimum for env-logger)).
I've also added the [rust-version](https://doc.rust-lang.org/cargo/reference/manifest.html#the-rust-version-field) field to the Cargo.toml to restrict the msrv (which also shows up on the crates.io site nicely).

File: Cargo.toml [hunk 1]
```diff
@@ -3,6 +3,7 @@ name = "mdns-sd"
 version = "0.8.0"
 authors = ["keepsimple <keepsimple@gmail.com>"]
 edition = "2018"
+rust-version = "1.60.0"
```
Comment (keepsimple1) [inline Cargo.toml:6]: To make thing simpler, let's use `1.63.0` here as well.

File: README.md [hunk 2]
```diff
@@ -32,7 +32,7 @@ Currently this library has the following limitations:
 
 ## Minimum Rust version
 
-Tested against Rust 1.61.0
+The Minimum Supported Rust Version is 1.60.0, running tests requires 1.63.
```
Comment (keepsimple1) [inline README.md:35]: Like mentioned above, we can just move to 1.63.0

**Review by keepsimple1 — COMMENTED** [review]
Thanks for your PR! Let's move the whole crate to rust 1.63, to simplify things.

**Comment by keepsimple1** [comment]
Sorry, for second thought, I think it is better to remove the dev-dependency of `env_logger`.  It is actually only used in `examples/` .  I think we don't really need it and a typical Rust user can always enable logging output using their favorite logger. Does it make sense?

**Review by keepsimple1 — APPROVED** [review]
Looks good! Thank you for updating!

**Pull request merged**